docs/codeql/conf.py

```diff
@@ -50,9 +50,9 @@
 import sys
 
 def setup(sphinx):
-    sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))	
+    sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
     from qllexer import QLLexer
-    sphinx.add_lexer("ql", QLLexer())
+    sphinx.add_lexer("ql", QLLexer)
```

Comment:
Try this. The [callable](https://docs.python.org/3/library/functions.html#callable) builtin should tell us whether an object is indeed callable. This way we can remain robust to different versions of Sphinx + pygments being installed, without breaking either local or CI usage.
```suggestion
    sphinx.add_lexer("ql", QLLexer() if callable(QLLexer) else QLLexer)
```

There are actually 3 places in the code we make this call:
- `docs/codeql/conf.py` (here)
- `docs/codeql/ql-training/conf.py`
- `documentation/restructuredtext/codeql-cli/conf.py` within the internal repo

Would you mind updating the first 2 locations in this PR, and then we can tackle the third in an internal PR?

Reply:
Thanks for the suggestion! Unfortunately, this fails with:
```
Exception occurred:
  File "/opt/homebrew/lib/python3.10/site-packages/sphinx/highlighting.py", line 130, in get_lexer
    lexer = lexer_classes[lang](**opts)
TypeError: 'QLLexer' object is not callable
```
I think the problem may be that `QLLexer` is callable regardless of the Sphinx version used. It is just `add_lexer` that has different expectations of how you pass the lexer in, depending on the version.

Reply:
Hmph yes good point. Thank you for trying. I wonder if we can detect from within Python what the Sphinx version is? If not I think the principled solution is the one in the internal issue - pin to a specific version of the packages.